daemon/create.go

```diff
@@ -16,6 +17,13 @@ func (daemon *Daemon) ContainerCreate(name string, config *runconfig.Config, hos
 		return "", warnings, err
 	}
 
+	// The check for a valid workdir path is made on the server rather than in the
+	// client. This is because we don't know the type of path (Linux or Windows)
+	// to validate on the client.
+	if config.WorkingDir != "" && !filepath.IsAbs(config.WorkingDir) {
```

Comment:
Been digging further and under the debugger. Runconfig.parse is client-side, not called on the daemon path which is where we want to do the check. I believe the correct place to put the check is where it currently is - at a relatively high layer in the daemon code path that is called during both run and exec. I could put it in runconfig ContainerConfigFromJob, but that further scatters checks. Or I could put it lower down in the exec driver itself, but that would mean a much longer code path before validation occurs, so that makes no sense. As ContainerCreate() already has other checks (memory), even though these are the hostconfig rather than config, this seems an appropriate location for the validation.